core/slim_globals.h

```diff
@@ -354,6 +354,432 @@ extern EidosValue_String_SP gStaticEidosValue_StringT;
+"                    \"description\": \"Whether the simuation was nucleotide-based.\",\n"
+"                    \"type\": \"boolean\"\n"
+"                },\n"
+"                \"separate_sexes\": {\n"
+"                    \"description\": \"Whether the simuation had separate sexes.\",\n"
+"                    \"type\": \"boolean\"\n"
+"                },\n"
+"                \"spatial_dimensionality\": {\n"
+"                    \"description\": \"The spatial dimensionality of the simulation.\",\n"
+"                    \"enum\": [\n"
+"                        \"\",\n"
+"                        \"x\",\n"
+"                        \"xy\",\n"
+"                        \"xyz\"\n"
+"                    ],\n"
+"                    \"type\": \"string\"\n"
+"                },\n"
+"                \"spatial_periodicity\": {\n"
+"                    \"description\": \"The spatial periodicity of the simulation.\",\n"
+"                    \"enum\": [\n"
+"                        \"\",\n"
+"                        \"x\",\n"
+"                        \"y\",\n"
+"                        \"z\",\n"
+"                        \"xy\",\n"
+"                        \"xz\",\n"
+"                        \"yz\",\n"
+"                        \"xyz\"\n"
+"                    ],\n"
+"                    \"type\": \"string\"\n"
+"                }\n"
+"            },\n"
+"            \"type\": \"object\"\n"
+"        }\n"
+"    },\n"
+"    \"type\": \"object\"\n"
+"}";
+
+const std::string SLIM_TSK_EDGE_METADATA_SCHEMA = "";
+const std::string SLIM_TSK_SITE_METADATA_SCHEMA = "";
+
+const std::string SLIM_TSK_MUTATION_METADATA_SCHEMA =
+"{\n"
+"    \"$schema\": \"http://json-schema.org/schema#\",\n"
+"    \"additionalProperties\": false,\n"
+"    \"codec\": \"struct\",\n"
+"    \"description\": \"SLiM schema for mutation metadata.\",\n"
+"    \"examples\": [\n"
+"        {\n"
+"            \"mutation_list\": [\n"
+"                {\n"
+"                    \"mutation_type\": 1,\n"
+"                    \"nucleotide\": 3,\n"
+"                    \"population\": 0,\n"
+"                    \"selection_coeff\": -0.2,\n"
+"                    \"slim_time\": 243\n"
+"                }\n"
+"            ]\n"
+"        }\n"
+"    ],\n"
+"    \"properties\": {\n"
+"        \"mutation_list\": {\n"
+"            \"items\": {\n"
+"                \"additionalProperties\": false,\n"
+"                \"properties\": {\n"
+"                    \"mutation_type\": {\n"
+"                        \"binaryFormat\": \"i\",\n"
+"                        \"description\": \"The index of this mutation's mutationType.\",\n"
+"                        \"index\": 1,\n"
+"                        \"type\": \"integer\"\n"
+"                    },\n"
+"                    \"nucleotide\": {\n"
+"                        \"binaryFormat\": \"b\",\n"
+"                        \"description\": \"Whether this mutation is a 'nucleotide-based' mutation.\",\n"
+"                        \"index\": 5,\n"
```

Comment:
It is the position, and it would be less confusing to sort by index, but we want to make it easy to output canonical JSON, and sorting on some arbitrary key is not an option in either the json library we're using or in python's json library. Sigh.

Reply:
See extensive discussion around https://github.com/tskit-dev/tskit/issues/763

Reply:
I see, ok